crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -697,7 +701,28 @@ where
 {
     ignore_then_commit(keyword(Keyword::Assert), parenthesized(expr_parser))
         .labelled(ParsingRuleLabel::Statement)
-        .map(|expr| Statement::Constrain(ConstrainStatement(expr)))
+        .map(|expr| {
+            let true_literal =
+                Expression::new(ExpressionKind::Literal(Literal::Bool(true)), expr.span);
+            Statement::Constrain(ConstrainStatement(expr, true_literal))
+        })
+}
+
+fn assertion_eq<'a, P>(expr_parser: P) -> impl NoirParser<Statement> + 'a
+where
+    P: ExprParser + 'a,
+{
+    ignore_then_commit(
+        keyword(Keyword::AssertEq),
+        parenthesized(expr_parser.separated_by(just(Token::Comma)).exactly(2)),
```

Comment:
Should probably allow_trailing here for consistency with other function arguments.

Also, nitpick but can you break this up a bit into a variable so the formatter doesn't do weird things with putting `ignore_then_commit` on its own line?